### PR TITLE
Fast json test

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
Quick experiment to see what the fast json gem looks like...

Using this [new serializer](https://github.com/openfoodfoundation/openfoodnetwork/pull/6886/commits/6b33aee4cfe949181cd38acfd0dafbd6a4ce9144), the new output looks roughly like this:

```
{"data"=>
  {"id"=>"391",
   "type"=>"customer",
   "attributes"=>
    {"id"=>391,
     "enterprise_id"=>1922,
     "name"=>nil,
     "code"=>"80lwfgefwoXAslUQcSO9leVYUgiKsaejyE2CfB4XQZgnTHF4Y/oA3sN4bJlFoA+zacOKTSP",
     "email"=>"devona.feeney@bernhard.biz",
     "allow_charges"=>false},
  "relationships"=>
    {"enterprise"=>{"data"=>{"id"=>"1922", "type"=>"enterprise"}}}}}
```

As opposed to this (roughly what it currently looks like):

```
 {"id"=>391,
  "enterprise_id"=>1922,
  "name"=>nil,
  "code"=>"80lwfgefwoXAslUQcSO9leVYUgiKsaejyE2CfB4XQZgnTHF4Y/oA3sN4bJlFoA+zacOKTSP",
  "email"=>"devona.feeney@bernhard.biz",
  "allow_charges"=>false}
```

### CODE REVIEW

File: app/assets/javascripts/admin/bulk_product_update.js.coffee [hunk 1]
```diff
@@ -147,7 +147,7 @@ angular.module("ofn.admin").controller "AdminProductEditCtrl", ($scope, $timeout
     if confirm("Are you sure?")
       $http(
         method: "DELETE"
-        url: "/api/products/" + product.id
+        url: "/api/legacy/products/" + product.id
```
Comment (jibees) [inline app/assets/javascripts/admin/bulk_product_update.js.coffee:150]: I'm wondering if there is an opportunity to have kind of builder somewhere like 
```javascript
function apiRoute(path) {
  return "/api/legacy/" + path; // something more robust and tested :p
}
```
But I know it should not changed very often.

**Comment by andrewpbrett** [comment]
@Matt-Yorkley I pushed some changes to make some specs pass. Hope you don't mind 😁

**Comment by andrewpbrett** [comment]
I also pushed https://github.com/openfoodfoundation/openfoodnetwork/pull/6886/commits/0a739777909982fd6bf864353c18e34f1bffe45a, in order to make it theoretically seamless for anyone who's using the existing endpoints to gradually transition off of them. I was sketching out a transition that looks something like:

1. `/api/*` redirects to `/api/legacy/*`
2. OFN frontend starts consuming `/api/legacy/*`

(Both of these are done by this PR)

3. We have a clean slate to build out `/api/beta/*` (or `/api/v1/*`, whatever) however we think is the way we want to go forward, **without having to worry about anyone currently consuming them, including the OFN front end**. Here you've started with JSON:API and FastSerializer 👍. 
4. We let anyone who's using the API (and getting redirected to the `/api/legacy/*` endpoints that soon, they'll start getting redirected to the `/api/v1/*` endpoints, and that their apps may break when that happens. But! They have the option of updating their apps/zaps/whatevers to consume the `/api/legacy/*` endpoints, for as long as we want to keep supporting them. We'd strongly encourage them to switch to the new ones. 
5. We can now slowly rip out any calls that the OFN frontend makes to `/api/legacy/*` as we rewrite things with Reactive Rails, and/or use the new `/api/v1/*` endpoints as necessary. Once that's done and everyone else has stopped using it, we can rip it out 🔥 
6. We're left with shiny, versioned, officially supported, super fast endpoints in `/api/v1/*` 🎉

**Comment by Matt-Yorkley** [comment]
This was more of an exploration rather than something I think we should merge, but we'll see. Some interesting things came out of it though.

When going through this process of moving all the existing bits to `/legacy` and then copying things into `/beta` one-by-one (and rewriting the serializers), it's really clear that there are lots of bits in our current controllers and serializers that we would almost certainly **not** want to replicate in the public API.

**Comment by andrewpbrett** [comment]
> This was more of an exploration rather than something I think we should merge

Noted. I just saw some failing tests and wanted to know if they'd block us, and the fastest way to find out was to fix them 🙂 

> we would almost certainly not want to replicate in the public API

+ 💯

**Comment by mkllnk** [comment]
This was just an experiment. Closing.

**Pull request closed without merging**